src/aap_eda/playbooks/project_clone.yml

```diff
@@ -0,0 +1,39 @@
+---
+# The following variables will be set by the runner of this playbook:
+# project_path: The destination folder
+# scm_url: https://server/repo
+# scm_branch: branch/tag/revision (HEAD if unset)
+# scm_refspec: a refspec to fetch in addition to obtaining version
+# key_file: private ssh file
+
+- hosts: localhost
+  gather_facts: false
+  connection: local
+  name: Update source tree if necessary
+  vars:
+    scm_branch: HEAD
+  tasks:
+    - name: Update project using git
+      tags:
+        - update_git
+      block:
+        - name: Update project using git
+          ansible.builtin.git:
+            dest: "{{ project_path | quote }}"
+            repo: "{{ scm_url }}"
+            version: "{{ scm_branch | default(omit) | quote }}"
+            refspec: "{{ scm_refspec | default(omit) }}"
+            key_file: "{{ key_file | default(omit) }}"
```

Comment:
@bzwei We should incorporate the GPG Key since that is just data flowing thru

Reply:
I will need to create another PR to include GPG key and test it out.